subprojects/docs/src/docs/userguide/authoring-builds/multi-project/multi_project_configuration_and_execution.adoc

```diff
@@ -0,0 +1,291 @@
+// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+[[configuration_and_execution]]
+= Configuration and execution
+
+<<build_lifecycle.adoc#sec:build_phases,Build phases>> describes the phases of every Gradle build.
+Let's zoom into the configuration and execution phases of a multi-project build.
+Configuration here means evaluating the build script file of a project, which includes downloading all plugins and build script dependencies.
+By default, the configuration of all projects happens before any task is executed.
+This means that when a single task, from a single project is requested, _all_ projects of a multi-project build are configured first.
+The reason every project needs to be configured is to support the flexibility of accessing and changing any part of the Gradle project model.
+
+
+[[sub:execution_time_dependencies]]
+== Execution dependencies
```

Comment:
I don't really get the point of this section. I think it is outdated and hence does not add value anymore. I think it can be removed.

We also never use the term "execution dependencies" except for this section. It's only confusing for the reaser.

As far as I understand the point of this section was to explain direct dependencies between tasks. But the sample in the "cross-project-configuration" section has changed. Because we don't want direct task dependencies to be used anymore.

Lets remove this one and the "firstMessag" sample snippet.

Reply:
Thanks! I'm always hesitant to remove sections so a confirmation that it is outdated/not useful is very good to have.

I am not entirely satisfied with what remains of the `multi_project_configuration_and_execution` page now. It firstly very briefly and rather unclearly talks about how all the projects are configured before any tasks are started (there are links from other places in the docs to this one), and then the remaining paragraphs (parallel execution, configure-on-demand, decoupled projects) seem rather unrelated. It'll need another iteration.

Reply:
I agree. I was also considering if this should be (re)moved completely. But I guess there is some interesting background knowledge here for users new to Gradle who want/need to understand what's going on "behind the scenes".

I think we can leave it like this for now, but do, as you say another iteration later.

Two possible ways forward:
- Revise/reword/restructure the section so that it fits better in and also clarify that this is deeper background knowledge you can safely skip if things work for you without it.
- Consider moving the content elsewhere into an "expert" part of the user manual (we might not have that place yet). It might make sense to combine it with the "build lifecycle" documentation we have elsewhere though.